src/modules/todo/api/index.ts

```diff
@@ -0,0 +1,15 @@
+import type { TodoItem } from '@/modules/todo/types/todo.types'
+
+type PostTodoData = Omit<TodoItem, 'id'>
+
+export const TodoApiService = {
+  retrieve: async (): Promise<TodoItem[]> => {
+    return fetch('api/todos')
+      .then(res => res.json() as Promise<{ todos: TodoItem[] }>)
+      .then(res => res.todos)
+  },
+
+  create: (data: PostTodoData): void => {
+    void fetch('/api/todos', { method: 'POST', body: JSON.stringify(data) })
```

Comment:
In general it's good idea to return request event from create and other data, as usually you will get newly created object in response that you might need. Also you could omit using return and just write fetch right away like this:

`create: (data: PostTodoData) => fetch('/api/todos', { method: 'POST', body: JSON.stringify(data) })`